### ISSUE TITLE
JSONConverter crashes

### ISSUE THREAD

**Issue by julifan** [spcl/perf-taint#30]
For the following output JSON (generated by the example from the "Accumulation of dependencies" issue), the assertion ` assert(pos != std::string::npos); ` in L. 430 fails. 

```
{
  "functions": {
    "f": {
      "file": "dependency-accumulation.c",
      "func_idx": 0,
      "line": 17,
      "loops": [
        {
          "callstacks": [
            [
              2
            ]
          ],
          "instance": {
            "0": {
              "level": 0,
              "params": [
                [
                  "x1"
                ]
              ]
            },
            "1": [
              {
                "entry_id": 0,
                "function_idx": 1
              }
            ]
          }
        }
      ]
    },
    "g": {
      "file": "dependency-accumulation.c",
      "func_idx": 1,
      "line": 9,
      "loops": [
        {
          "callstacks": [
            [
              2,
              0
            ]
          ],
          "instance": {
            "0": {
              "level": 0,
              "params": [
                [
                  "x2"
                ]
              ]
            }
          }
        }
      ]
    }
  },
  "functions_demangled_names": [
    "float",
    "__float128",
    "main",
    "register_variable"
  ],
  "functions_mangled_names": [
    "f",
    "g",
    "main",
    "register_variable"
  ],
  "functions_names": [
    "f",
    "g",
    "main",
    "register_variable"
  ],
  "parameters": [
    "x1",
    "x2"
  ],
  "unused_parameters": [
    "p"
  ]
}
```

**Comment by julifan** [comment]
I've attached both: 

[dependency-accumulation.zip](https://github.com/spcl/perf-taint/files/6918564/dependency-accumulation.zip)

**Comment by mcopik** [comment]
@julifan Thank you so much! I was able to reproduce the issue. In the end, this is a bug in demangle function from C++ ABI - it does not always produce correct output on a non-mangled name. I added a workaround similar to the one in LLVM, it should work on Linux.

Fixed on master in ded2139
